source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,147 @@
+==================
+Tuning Connections
+==================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. _connection-storms-java:
+
+Overview
+--------
+
+In this guide, you can learn about tuning connections to prevent **connection storms**,
+what connection storms are, and common causes.
+
+A **connection storm** is an event in which application servers
+start an increasing number of connection requests, but the host server
+CPU is unable to schedule them.
+
+.. example::
+
+   Consider an application that uses MongoDB, and is served from an
+   auto scaling service. A sudden spike in traffic could cause the scaling
+   service to stand up many new application instances. This sudden burst
+   in application instances attempting to establish connections to MongoDB
+   is one example of a connection storm.
+
+Common causes for connection storms in MongoDB include:
+
+- Incorrect application behavior
+- Network latency
+- Insufficient hardware for the workload
+- Failover events
+
+In MongoDB, connection storms can have a multiple effects. For example:
+
+- The high volume of connection requests leads to latency in processing.
+- Replication connections between nodes are blocked or slowed, so replication
```

Comment:
You'd have to check with the server team on this, but I believe improvements have been made in this area so that connection storms shouldn't block intra-cluster communication as much.